frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/common-templates-validations.ts

```diff
@@ -0,0 +1,98 @@
+import { getFieldId } from '../../../utils/vm-settings-tab-utils';
+import { UpdateOptions } from '../../types';
+import { iGetVmSettingAttribute } from '../../../selectors/immutable/vm-settings';
+import { iGetLoadedCommonData } from '../../../selectors/immutable/selectors';
+import { VMSettingsField, VMWizardProps, VMSettingsRenderableField } from '../../../types';
+import {
+  CommonTemplatesValidation,
+  IntegerValidationObj,
+  IntegerValidtionType,
+} from './validation-types';
+import {
+  getRelevantTemplates,
+  getValidationsFromTemplates,
+  getFieldValidations,
+  getValidationAttributes,
+  getBetweenIntegerValidationObj,
+  getMinIntegerValidationObj,
+  getMaxIntegerValidationObj,
+} from './selectors';
+
+// TODO: Add all the fields in the form
+// For each field we need to check for validations
+const IDToJsonPath = {
+  [getFieldId(VMSettingsField.MEMORY)]: '.spec.domain.resources.requests.memory',
+  [getFieldId(VMSettingsField.CPU)]: '.spec.domain.cpu.cores',
+};
+
+export const getTemplateValidations = (
+  options: UpdateOptions,
+  fieldId: VMSettingsRenderableField,
+): CommonTemplatesValidation[][] => {
+  // Proceed if there are no validation for the specific fieldId
+  if (!(fieldId in IDToJsonPath)) {
+    return [];
+  }
+  const { getState, id } = options;
+  const state = getState();
+
+  // Get OS, workload-profile and flavor attributes
+  const os = iGetVmSettingAttribute(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const flavor = iGetVmSettingAttribute(state, id, VMSettingsField.FLAVOR);
+  const workloadProfile = iGetVmSettingAttribute(state, id, VMSettingsField.WORKLOAD_PROFILE);
+
+  // Get all the templates from common-templates
+  const commonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+
+  // Get templates based on OS, WorkloadProfile and Flavor
+  const relevantTemplates = getRelevantTemplates(commonTemplates, os, workloadProfile, flavor);
+
+  // Get all the validations from the relevant templates
+  const validations = getValidationsFromTemplates(relevantTemplates);
+
+  // Return all the validations which are relevant for the field
+  return getFieldValidations(validations, IDToJsonPath[fieldId]);
+};
+
+export const getIntegerValidationType = (
+  validations: CommonTemplatesValidation[],
+): IntegerValidationObj => {
+  /*
+  Integer validation can be one of three options:
+  1. minimum: only 'min' rule | validation = [{min: 1}, ...]
+  2. maximum: only 'max' rule | validation = [{max: 10}, ...]
+  3. between: both 'min' and 'max' rules | validation = [{min: 1}, {max: 10}, ...] or [{min: 1, max: 10}, ...]
+  */
+  let integerValidationObj: IntegerValidationObj;
+  // eslint-disable-next-line consistent-return
+  validations.forEach((validation) => {
+    const validationAttrs = getValidationAttributes(validation);
+    if (validationAttrs.includes('min') && validationAttrs.includes('max')) {
```

Comment:
can we check the object directly without getting the `validationAttrs` ? 
e.g. is their a way to ask if `validation` itself has an attribute without first getting all the attributes into a list ... ?

`getValidationAttributes` looks like something that we should get for free from the language ...

Reply:
+1

Reply:
Right. I'll remove the method